citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -375,7 +379,17 @@ public CitationResults getGroupItems(IUser user, String groupId, String collecti
                 total = citations.size();
             }
         }
-        results.setCitations(citations != null ? citations : new ArrayList<>());
+        HashSet<String> hiddenItemSet = asyncCitationManager.getHiddenItemsList();
+        List<ICitation> finalCitations = new ArrayList<>();
+        for(ICitation citation : citations) {
+            if(!hiddenItemSet.contains(citation.getKey())) {
+                finalCitations.add(citation); 
+            }
+            else {
+                total--;
+            }
+        }
```

Comment:
just use the stream API and filter by hidden status.

Reply:
or actually, you probably want to only retrieve the citations from the db that are not hidden.